### ISSUE TITLE
Automatically enable volume envelope whenever changing default envelope

### ISSUE THREAD

**Issue by Monospace-V** [LMMS/lmms#6121]
### Enhancement Summary
Automatically turn AMT to 100 whenever a volume envelope is changed from the default.

#### Justification
New users to LMMS often do not realise the amount knob has to be turned. If AMT automatically moves to full when an envelope is changed from default, then we have a double advantage:
a) user need not look up how to enable envelope etc
b) user notices AMT knob going to hundred and can realise what it does.

**Comment by Spekular** [comment]
This should only occur if the amount is at zero before tweaking, but otherwise it seems like a reasonable enhancement. It's not too different from the way the EQ enables bands when they're modified.

**Comment by RoxasKH** [comment]
> 
> ### Enhancement Summary
> 
> Automatically turn AMT to 100 whenever a volume envelope is changed from the default.
> #### Justification
> 
> New users to LMMS often do not realise the amount knob has to be turned. If AMT automatically moves to full when an envelope is changed from default, then we have a double advantage:
> a) user need not look up how to enable envelope etc
> b) user notices AMT knob going to hundred and can realise what it does.

I mean, you can click on the envelope window and the amount will automatically get to 100%.
Being grey it's not difficult to understand it's disabled, also considering it does nothing.

But i guess i really haven't seen anyone making changes on the envelope without having it active.

Although, here's why it wouldn't work: the amount knob exists for a reason, even if it's u ncommon you may want to have a 50% amount on your envelope.

Now imagine you wanna make a change on your 50%amt envelope: you're doing the change and the amount reset to 100%, and you need to bring it again to 50%; that wouldn't be very userfriendly, and quite annoying.

FL Studio doesn't have an amount knob on the volume envelope.
Although it has it on other envelopes, and they do it more smartly:

- First: the amount is being visually drawn as well as the other parameters as the height of the whole line
- Second: the amount is initially set on 0, but it also uses a toggle that enable/disable the envelope, and that toggle is being put on whenever you do a change.

![immagine](https://user-images.githubusercontent.com/40141826/128050612-4e4c1f98-7935-4fdc-851c-dc3151779527.png)


A way to fix the problem would be adding the same kind of toggle on LMMS envelope and setting the Amount on 100% by default, and the toggle on Off by default. Whenever you do a change, the toggle will be put on, and the amount would be on 100% by default: if you want a different amount you can still change it, and disable the envelope using the toggle.

It's also true that FL probably uses a defult envelope when the actual envelope is disabled to avoid the clicks LMMS has.
And i still think LMMS could follow the same path, as this default envelope (so with sustain up all the way and a bit of release, the other parameters aren't really relevant to avoid them) when disabled would fix any clicks caused by the program and not the samples.

![immagine](https://user-images.githubusercontent.com/40141826/128050756-210c917c-8e24-499b-b150-b59db7add03a.png)

**Comment by Monospace-V** [comment]
Roxas, that is completely logical, but logical != intuitive. Why would someone think of clicking the envelope when the changes didn't work? Generally, if their changing attack doesn't work, they turn everything up and struggle to listen for any change. After you know the method, it seems obvious, but I've seen many new users struggling to change their envelope.

**Comment by Spekular** [comment]
> Although, here's why it wouldn't work: the amount knob exists for a reason, even if it's u ncommon you may want to have a 50% amount on your envelope.
> 
> Now imagine you wanna make a change on your 50%amt envelope: you're doing the change and the amount reset to 100%, and you need to bring it again to 50%; that wouldn't be very userfriendly, and quite annoying.

This is easily solved by the check for amt = 0.

**Comment by RoxasKH** [comment]
> 
> 
> > Although, here's why it wouldn't work: the amount knob exists for a reason, even if it's u ncommon you may want to have a 50% amount on your envelope.
> > Now imagine you wanna make a change on your 50%amt envelope: you're doing the change and the amount reset to 100%, and you need to bring it again to 50%; that wouldn't be very userfriendly, and quite annoying.
> 
> This is easily solved by the check for amt = 0.

That's true, you're right. Although having a separate toggle could allow to disable the envelope remembering the amount knob value. Under the current implementation, the only way to disable it would be to set the amount to 0 again, and if you need to bring it back, you'd have to remember the value the amount knob had previously.

**Comment by musikBear** [comment]
> It's also true that FL probably uses a defult envelope when the actual envelope is disabled to avoid the clicks LMMS has.

I implemented a default ENV for all new _my-Instruments_ instances. It prevent clicks.
Not sure it it 'made' it into 1.3 though. May still be in 'review hiatus'

**Comment by zonkmachine** [comment]
> Not sure it it 'made' it into 1.3 though. May still be in 'review hiatus' 

All of your PR's has been closed and not merged into lmms.

**Comment by bulbul-akabulbul** [comment]
I'll get to working on it!

**Comment by michaelgregorius** [comment]
I was inspired by [this video](https://youtu.be/XyrCwuU43Qc) and some of the comments to do something about the clicky sounds in LMMS and found this issue after my fix was done.

In the video the clicky sounds are first mentioned [here](https://youtu.be/XyrCwuU43Qc?t=63) and a bit later [here](https://youtu.be/XyrCwuU43Qc?t=145). This seems to be an experienced producer who could not find out how to solve the problem. You can see one attempt [here](https://youtu.be/XyrCwuU43Qc?t=104) where he's increasing the release but with a turned off "Amount".

Some comments which document the confusion and the probem well can be found [here](https://www.youtube.com/watch?v=XyrCwuU43Qc&lc=UgzO_v6xHYiVuy1SG3F4AaABAg) and [here](https://www.youtube.com/watch?v=XyrCwuU43Qc&lc=Ugx3vr_OKMElwkr8UuN4AaABAg).

This comment especially caught my eye: "Why won't the lmms team just straight fix these problems people have been compaining about for years and make it usable out of the box? It's a mere case of config so I see no reason not to fix them." I have to admit that this person is not really wrong.

Another comment seems to indicate that some people never find out how to get rid of the clicks: "Omg thank you so much, the clicking always fed me up".

## The fix
The fix is to simply activate the volume envelope by default for instruments that are added to the project as well as the instruments in the default project.

Furthermore the following changes are made to the envelopes:
* The amount of the volume envelope is set to 1, i.e. it is activated by default.
* The amounts of the cutoff and resonance envelopes are kept at 0 so that users have to opt-in to them. This might also potentially save resources.
* The amount for sustain is changed from 0.5 to 1. This way users start with a full sustained sound.
* The amount for hold is set to 0.

The reasoning for the changes is to enable users to quickly dial in some well known envelopes, especially if they only know ADSR envelopes and don't known how an AHDSR envelope works. A user that wants to dial in some pluck envelope might try to turn the sustain down to 0. However, if the "hold" parameter was still set to 0.5 by default it would get in the way and users had to learn that they have to turn the "hold" down to 0. With the new set of default values they can discover the "hold" feature without it getting into the way with simpler approaches.

See pull request #6864 for some more details.
